cshared/main.go

```diff
@@ -71,6 +71,31 @@ func getCertPem(configFilePath string) []byte {
 	return certChainPem
 }
 
+// Returns a string matching ECP's key type.
```

Comment:
Golang public function comments need to start with the function name (i.e. GetKeyType returns ...) Btw can we move this to the very end of the file so it doesn't come between GetCertPem and getCertPem? Also, it might also be worth revisiting the existing APIs. GetCertPemForPython already needs to be called twice, and now GetKeyType makes another call to instantiate client.Cred - this could be unnecessary overhead. Are we able to return key type and cert length in one call? Something to think about before we commit - once the APIs are released it will be hard to change.

Reply:
I prefer a separate API because I need the key type much earlier than the certificate, and it increases the complexity of getCert PEM.

I think this has a fairly low overhead, compared to the overall TLS handshake and sign operations.